pkg/controller/build/build_controller_test.go

```diff
@@ -511,5 +511,115 @@ var _ = Describe("Reconcile Build", func() {
 				Expect(reconcile.Result{}).To(Equal(result))
 			})
 		})
+
+		Context("when source URL is specified", func() {
+			// validate file protocol
+			It("fails when source URL is invalid", func() {
+				buildSample.Spec.Source.URL = "foobar"
+
+				// Fake some client LIST calls and ensure we populate all
+				// different resources we could get during reconciliation
+				client.ListCalls(func(context context.Context, object runtime.Object, _ ...crc.ListOption) error {
+					switch object := object.(type) {
+					case *corev1.SecretList:
+						list := ctl.FakeSecretList()
+						list.DeepCopyInto(object)
+					case *build.ClusterBuildStrategyList:
+						list := ctl.ClusterBuildStrategyList(buildStrategyName)
+						list.DeepCopyInto(object)
+					}
+					return nil
+				})
+
+				statusCall := ctl.StubFunc(corev1.ConditionFalse, fmt.Sprintf("invalid source url"))
```

Comment:
No need for `Sprintf` if there is nothing to format.
```suggestion
				statusCall := ctl.StubFunc(corev1.ConditionFalse, "invalid source url")
```